api/controllers/HistoryController.js

```diff
@@ -25,18 +25,78 @@
  * @help        :: See http://sailsjs.org/#!/documentation/concepts/Controllers
  */
 
-/* global MachineService, JsonApiService */
+/* global MachineService, JsonApiService, History */
 
 module.exports = {
   create(req, res) {
     req.body = JsonApiService.deserialize(req.body);
 
-    if (req.body.data.attributes['end-date'] === '') {
-      MachineService.sessionOpen(req.user);
-    } else {
-      MachineService.sessionEnded(req.user);
+    return this.createRecord(req, res);
+  },
+
+  createRecord(req, res) {
+    var data = req.body.data.attributes || {};
+
+    if (req.allParams().id) {
+      data.id = req.allParams().id;
+    }
+    History.create(data)
+      .then( (newInstance) => {
+
+        return History.findOne(newInstance.id)
+          .then((newRecord) => {
+
+            return res.created(newRecord);
+          })
+          .catch((err) => {
+            return res.negotiate(err);
+          });
+      })
+      .then(() => {
+        if (data.endDate === '') {
+          MachineService.sessionOpen(req.user);
+        }
+      })
+      .catch((err) => {
+        console.log('error');
+        console.log(err);
+      });
+  },
+
+  update(req, res) {
```

Comment:
Does not guacamole only send POST request for both history addition and creation ?